standard/requirements/reqA01.adoc

```diff
@@ -7,12 +7,20 @@
 
 // {set:cellbgcolor:#FFFFFF}
 
-A netCDF file shall be provided along with its own identity, as a URI, external to the file contents.
+A netCDF file content root group shall be identified by an identity URI, external to the file contents. 
 
-If no identity is provided, then the path used by a library to access the file is used as a default identity.
+The identity of the root group shall always terminate in a '/' character.
+
+By convention, the prefix 'this:' for the identity URI may be used by implementations representing a single netCDF file within a graph.
+
+Implementations shall provide an optional 'uri' option to specify the identity of the file contents. This identity URI shall conform to the W3 URI specification (https://datatracker.ietf.org/doc/html/rfc3986). Note, this includes 'http:', 'https:', 'file:', 'urn:' URIs.
+
+Implementations shall provide an optional 'download-url' option. If no 'uri' is provided, but a 'download-url' is provided, then this shall be used as the file contents identity, ensuring that a trailing '/' is added if is does not exist in the 'download-url'.
```

Comment:
How does the download-url option differ from the uri option?

Reply:
The URI is intended to be the identifier for the root group. This identifies the first container, with the content of the file defined with respect to that identifier.

The root container must have an identifier.  If there is no identifier provided at run time, then one is inferred.

The download-url is optional, and enables a specific location to be specified where the binary file may be obtained from.  It is expected that the download-url is a resolveable netCDF file.

it is plausible that a download-url is provided, but that the file identity is differently structured.  The identity may be part of a managed naming scheme, whilst some entities within that naming scheme may be served from different uri patterns

```
uri = http://data.example.com/formalised/identification/scheme/identity-of-file/
download-url = http://data-portal.example.com/data-service/this-particular-file.nc
```

In this case, the download-url would only be used within the dcat:distribution blank node, as defined in the new version of reqA02

```
this: a bald:Container ;
    dcat:distribution [
    	a dcat:Distribution;
       dcat:downloadURL <http://data-portal.example.com/data-service/this-particular-file.nc> ;
...
    ] .
```

There's a case, which I thought to explore here: which is whether it is useful to specify that `download-url` could be specified alone, without a `uri`, and that this could then be used to infer URI, e.g.

```
download-url = http://data.example.com/formalised/identification/scheme/identity-of-file.nc
```
therefore URI is inferred to be
```
URI = http://data.example.com/formalised/identification/scheme/identity-of-file.nc/
```
(with added trailing '/' as per the specification)

So, 
* URI must exists, and will be inferred if not specified, but if specified by `uri`, this is used.
* downloadURL is optional and shall be omitted from the output graph if not provided.
    * if provided, but `uri` not specified, then it shall be used to infer URI in preference to a local `file:/` identifier
    * this raises a further implementation questions: if an implementation supports remote graph creation, does this remote identifier get interpreted as a download-url - this could be a neat interpretation, but is not mandated at present.

@adamml @jyucsiro  any thoughts you have on this would be very welcome